x/profile/types/genesis.go

```diff
@@ -25,6 +28,62 @@ func (gs GenesisState) Validate() error {
 
 	// this line is used by starport scaffolding # genesis/types/validate
 
+	if err := gs.validateValidator(); err != nil {
+		return err
+	}
+
+	return gs.validateCoordinator()
+}
+
+func (gs GenesisState) validateValidator() error {
+	// Check for duplicated index in consensusKeyNonce
+	consensusKeyNonceIndexMap := make(map[string]string)
+	for _, elem := range gs.ConsensusKeyNonceList {
+		index := string(ConsensusKeyNonceKey(elem.ConsAddress))
+		if _, ok := consensusKeyNonceIndexMap[index]; ok {
+			return fmt.Errorf("duplicated index for consensusKeyNonce: %s", elem.ConsAddress)
+		}
+		consensusKeyNonceIndexMap[index] = elem.ConsAddress
+	}
+
+	// Check for duplicated index in validatorByConsAddress
+	validatorByConsAddressIndexMap := make(map[string]string)
+	for _, elem := range gs.ValidatorByConsAddressList {
+		index := string(ValidatorByConsAddressKey(elem.ConsAddress))
+		if _, ok := validatorByConsAddressIndexMap[index]; ok {
+			return fmt.Errorf("duplicated index for validatorByConsAddress: %s", elem.ConsAddress)
+		}
+		validatorByConsAddressIndexMap[index] = elem.ConsAddress
+
+		// Remove to check if all consensus key exist
+		delete(consensusKeyNonceIndexMap, index)
+	}
+	// Check if all consensus key exist
+	for _, address := range consensusKeyNonceIndexMap {
+		return fmt.Errorf("consesus key not found for address: %s", address)
```

Comment:
The error message mentions "address" and the variable is `consensusKeyNonceIndexMap`. Which one do you want to check?

We actually don't need to check if `consensusKeyNonce` is associated with a `validatorByConsAddress` because theoretically `validatorByConsAddress` could be set in the genesis without signing a proof and so without having to set `consensusKeyNonce` and we can have a nonce value for the signature without an existing consensus pub key in the store.

So `consensusKeyNonce`. can be set in the genesis totally independently from the other values

Reply:
Concerning validator address:
A consensus key must be associated with validator address but the opposite is actually not true.
Validator profiles are used for:
- Giving a description of the validator
- Matching a consensus address on a launched blockchain to a validator address on SPN

Since a validator profile may be used to just provide a description of the validator, we can have a validator that is not associated with a consensus address

Although if the consensus address value is determined for this validator, then it must exist in the store